src/main/java/seedu/address/logic/commands/MailCommand.java

```diff
@@ -41,28 +42,27 @@ public CommandResult execute(Model model, CommandHistory history) throws Command
         if (model.getFilteredPersonList().isEmpty()) {
             return new CommandResult(String.format(MESSAGE_EMAIL_CONTACT_EMPTY));
         }
-        StringBuilder builder = new StringBuilder();
-        for (int i = 0; i < model.getFilteredPersonList().size(); i++) {
-            if (i == model.getFilteredPersonList().size() - 1) {
-                builder.append(model.getFilteredPersonList().get(i).getEmail());
-            } else {
-                builder.append(model.getFilteredPersonList().get(i).getEmail()).append(",");
-            }
-        }
+        String emailList = model.getFilteredPersonList().stream()
+                .map(person -> person.getEmail().toString())
+                .collect(Collectors.joining(","));
         try {
-            Desktop desktop;
-            if (Desktop.isDesktopSupported()
-                    && (desktop = Desktop.getDesktop()).isSupported(Desktop.Action.MAIL)) {
-                URI mailto = new URI("mailto:" + builder);
-                desktop.mail(mailto);
-            } else {
-                throw new RuntimeException("Desktop doesn't support mailto");
-            }
+            openMailApp(emailList);
         } catch (URISyntaxException | IOException e) {
-            throw new RuntimeException("Error");
+            throw new RuntimeException(e.getMessage());
```

Comment:
I don't know if you really have a rationale for the `RuntimeException`, we can't really show an error message at this point so can't really do much anyways. But the ideal case would've been to throw a `CommandException`.

Reply:
This is mostly just me wondering out loud. We can have this in planned enhancements.